integration-cli/docker_cli_update_unix_test.go

```diff
@@ -187,3 +176,25 @@ func (s *DockerSuite) TestUpdateStats(c *check.C) {
 	c.Assert(preMemLimit, checker.Equals, curMemLimit)
 
 }
+
+func (s *DockerSuite) TestUpdateRestartPolicy(c *check.C) {
+	out, _ := dockerCmd(c, "run", "-d", "--restart=on-failure:3", "busybox", "sh", "-c", "sleep 1 && false")
+	timeout := 15 * time.Second
+	if daemonPlatform == "windows" {
+		timeout = 50 * time.Second
+	}
+
+	id := strings.TrimSpace(string(out))
+
+	// update restart policy to on-failure:5
+	dockerCmd(c, "update", "--restart=on-failure:5", id)
+
+	err := waitInspect(id, "{{ .State.Restarting }} {{ .State.Running }}", "false false", timeout)
```

Comment:
This one is flakey because the container is not going to be running for a very long time so it may be difficult to catch (And is probably why it's currently failed on gccgo).

Maybe checking just the `RestartCount` hits 5, like below, and then verify that it is in a stopped state.